### ISSUE TITLE
Context menu bug Google Chrome

### ISSUE THREAD

**Issue by raymiec** [jaedb/spotmop#87]
Upon selection of a track via checkbox or right click the context menu shows for a second and quickly hides itself again. thus the context options can't be clicked.  This behaviour presents itself on windows using chrome and also on android. behaves as expected in firefox.

**Comment by jaedb** [comment]
@raymiec This is a known issue and was tied to the 'Copy URI' functionality introduced in `2.6.4`. It'll be resolved in the next release (currently in dev).

Just to confirm - you're experiencing this on touch devices only?

**Comment by raymiec** [comment]
Yeah both are touch devices, although the laptop is a hybrid (has a touch screen), i disabled the hid drivers, and rebooted, the issue still presents itself on chrome, i'll check it later on a normal laptop and report back.   For now i moved the context menu div.

```
<div id="body" class="body animate">
        <div ui-view></div>
        <contextmenu></contextmenu>
    </div>
```

it now functions on chrome in android, the menu appears static on the header of the page, is this expected behaviour? would be better if it floated at the top of the screen instead of having to scroll to the top to execute actions, kinda defies the point of a context menu i.m.o.     The script itself is lovely. much appreciated.

**Comment by jaedb** [comment]
It's weird that moving the markup changed anything (I broke it on environment).

I suspect the browser was detecting (rightly or wrongly) a scroll event, and hid the context menu. This doesn't make for a friendly UX on touch devices in any case, so I've limited it to non-touch devices.

I've just pushed this to `develop` branch now (commit 2707b70). Are you able to confirm that this fixes the issue?

**Comment by raymiec** [comment]
Sorry i removed the scroll event, the js file was cached so thought it was just moving the contextmenu div.   What negative impact is removing that event likely to have?

**Comment by raymiec** [comment]
yeah looking at your commit that's what i removed and it fixed the issue on mobile.  on the desktop browser right click is not working and neither is when you check the boxes in the list of tracks, if you resize the browser to mobile size then selection of the checkboxes work. although if you proceed to right click it totally screws it again.

**Comment by jaedb** [comment]
This appears to be an issue with a hybrid device. I don't have such a device to test on, so it'll be tricky to accurately patch this.

It sounds like you're pretty handy with HTML + Javascript, perhaps you could have a look into this?

**Comment by raymiec** [comment]
Just tested it on my other laptop, not hybrid, similar specs,  windows 10, latest ver of chrome. The problem still persists. I'll look into it and hopefully get a proper fix for it soon. I'm going to get in about the project over the weekend. I've not worked with Angular.js before so once i get my head round that i may be able to contribute more.  

Also a bit offtopic for this thread sorry but there's a modification i would like to make to it,  a kick in the right direction would be appreciated. I want to add a separate page to the left menu, that fetches a json string of spotify tracks from another server and populates the a listview and gives me access to the same functionality the app has already, i.e right click to add to queue etc..  Like i said a kick in the right direction would be appreciated when working with angular,  if its something you wouldn't mind coding yourself as a separate addon for me i would happily send a little something to your paypal account.   Thanks

**Comment by Bluscreenz** [comment]
Same issue on iPhone in both Chrome and Safari.

**Comment by raymiec** [comment]
I'm just waiting on a new micro sd card being delivered for my raspberry pi and i'll get another installation built and look into the issue properly.

**Comment by gaiterjones** [comment]
This fixes the context menu issue on i devices : 
 app.js 30471

```
// still no context menu? then we have clicked outside of one, so hide 'em
if( !contextMenu.is('contextmenu') && !$rootScope.isTouchDevice()){
    $rootScope.$broadcast('spotmop:contextMenu:hide');
}
```

**Comment by raymiec** [comment]
@gaiterjones confirmed fix on android/chrome too. nice one.

to tie the whole context menu thing up, we need to disable right click on touch devices and when emulated touch is active. at present when you right click on a touch device part of the context menu will appear with none of its formatting properties which also breaks the context menu check selecting a checkbox.

**Comment by jaedb** [comment]
Awesome work guys, I've found it quite tricky as some devices seem to work fine with this functionality whilst others experience this phantom context menu issue.

I'd like to maintain the functionality where the context menu is hidden when you click anywhere outside of the context menu (expected UX). It's just a matter of working it out across all devices.

In the meantime, I've changed the _Emulate Touch Device_ setting. It's now a radio button to allow the user to govern how their hybrid device behaves. Have a peep on the `develop` branch. It'd be great if you could have a quick spin on a hybrid device and let me know if this works for you.

**Comment by raymiec** [comment]
Nice one good work, i gave it a shot, works as expected. tested on

normal windows 10 laptop, firefox & chrome,
hybrid laptop windows 8.1, firefox and chrome
Android Chrome.

i prefer the context menu being static at the bottom as well.

Couple of side notes, if you get time will you look into the bug with the android webview, not being able to authenticate spotify, the injected iframe listens for a postmessage, in the chrome browser its fine but within a webview container the value isn't passed, returns null.

also how did you get on with running mopidy on osmc ?

https://discourse.osmc.tv/t/mopidy-on-osmc-alpha-4/456/2

**Comment by raymiec** [comment]
another couple of minor issues,  if you have a list on items in the queue, you can select multiple  as expected, navigate away from the page and go back and now it will only let you select one item at a time.

When on mobile or if i resize the browser when i try re order the queue an overlay appears from the side with options to add to library, playlist, queue, this is fine on desktop but not needed on mobile, plus the formatting is a mess, half of it is clipped.

**Comment by jaedb** [comment]
@raymiec Can you please create a new issue for the `postmessage` error in Android webview?

Regarding the other issues, I've patched the bug where multi-select stopped working on the _Now Playing_ tracklist when navigating away. This was due to an orphaned `$rootScope` listener.

These patches will be included in the next release which will be coming out shortly.
